didel/student.py

```diff
@@ -44,13 +44,46 @@ def populate(self, soup, *args, **kw):
             setattr(self, attr, value)
 
 
-    @login_required
     def get_course(self, ref):
         """
         Return a Course object
         """
+        if not self.logged:
+            raise DidelLoginRequired()  
         if not ref in self._courses:
             c = Course(ref, self)
             c.fetch(self.session)
             self._courses[ref] = c
         return self._courses[ref]
+
+    def get_all_courses(self):
+        """
+        return all courses (and keep them in self._courses with get_course(ref))
```

Comment:
Remove the precision in parentheses here, this is an internal feature that shouldn’t be in the API.